anvil/src/genesis.rs

```diff
@@ -0,0 +1,246 @@
+    pub mix_hash: Option<H256>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub coinbase: Option<Address>,
+    #[serde(default)]
+    pub alloc: Alloc,
+    #[serde(
+        default,
+        deserialize_with = "deserialize_stringified_u64_opt",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub number: Option<u64>,
+    #[serde(
+        default,
+        deserialize_with = "deserialize_stringified_u64_opt",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub gas_used: Option<u64>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub parent_hash: Option<H256>,
+    #[serde(
+        default,
+        deserialize_with = "deserialize_stringified_numeric_opt",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub base_fee_per_gas: Option<U256>,
+}
+
+impl Genesis {
+    /// Loads the `Genesis` object from the given json file path
+    pub fn load(path: impl AsRef<Path>) -> Result<Self, FsPathError> {
+        foundry_common::fs::read_json_file(path)
+    }
+
+    /// The clap `value_parser` function
+    pub(crate) fn parse(path: &str) -> Result<Self, String> {
+        Self::load(path).map_err(|err| err.to_string())
+    }
+
+    pub fn chain_id(&self) -> Option<u64> {
+        self.config.as_ref().and_then(|c| c.chain_id)
+    }
+
+    /// Applies all settings to the given `env`
+    pub fn apply(&self, env: &mut Env) {
+        if let Some(chain_id) = self.chain_id() {
+            env.cfg.chain_id = chain_id.into();
+        }
+        if let Some(timestmap) = self.timestamp {
+            env.block.timestamp = timestmap.into();
+        }
+        if let Some(base_fee) = self.base_fee_per_gas {
+            env.block.basefee = base_fee;
+        }
+        if let Some(number) = self.number {
+            env.block.number = number.into();
+        }
+        if let Some(coinbase) = self.coinbase {
+            env.block.coinbase = coinbase;
+        }
+        env.block.difficulty = self.difficulty.into();
+        env.block.gas_limit = self.gas_limit.into();
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Default)]
+#[serde(transparent)]
+pub struct Alloc {
+    pub accounts: HashMap<Address, GenesisAccount>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct GenesisAccount {
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub code: Option<Bytes>,
+    #[serde(default, skip_serializing_if = "HashMap::is_empty")]
+    pub storage: HashMap<H256, H256>,
+    #[serde(deserialize_with = "deserialize_stringified_numeric")]
+    pub balance: U256,
+    #[serde(
+        default,
+        deserialize_with = "deserialize_stringified_u64_opt",
+        skip_serializing_if = "Option::is_none"
+    )]
+    pub nonce: Option<u64>,
+    #[serde(rename = "secretKey", default, skip_serializing_if = "Option::is_none")]
+    pub private_key: Option<Bytes>,
```

Comment:
Not sure how complex this would be, but adding the private key to the node's keyring could be useful here. geth really only uses this for testing but could be useful for setting particular accounts in state at startup and then being able to send transactions from them via `eth_sendTransaction`